api/allocations.go

```diff
@@ -91,6 +91,7 @@ func (a *Allocations) Info(allocID string, q *QueryOptions) (*Allocation, *Query
 // long pauses on this API call.
 func (a *Allocations) Exec(ctx context.Context,
```

Comment:
To implement the equivalent for actions, I think we want to:
* Copy this to something `(*Job) ActionExec`
* Add a field to `execSession` to change the path that it makes the initial connection to (so that it points to our new HTTP API endpoint).

Reply:
the execSession field I ended up watching for is `action`; it modifies reqpath if present: https://github.com/hashicorp/nomad/pull/18778/files#diff-3735e959db9a859486511196a56619a36ce39c1ab4179a709ef6ccfd97d35d15R101-R106